File: app/src/main/java/in/hbb20/countrycodepickerproject/CustomColorFragment.java
package in.hbb20.countrycodepickerproject;

import android.graphics.PorterDuff;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.hbb20.CountryCodePicker;


/**
 * A simple {@link Fragment} subclass.
 */
public class CustomColorFragment extends Fragment {

    public CustomColorFragment() {
    }

    TextView textViewTitle;
    EditText editTextPhone;
    CountryCodePicker ccp;
    RelativeLayout relativeColor1,relativeColor2,relativeColor3;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_custom_color, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        assignViews();
        setClickListener();
    }

    private void setClickListener() {
        relativeColor1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setColor(1, getActivity().getResources().getColor(R.color.color1));
            }
        });

        relativeColor2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setColor(2, getActivity().getResources().getColor(R.color.color2));
            }
        });

        relativeColor3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setColor(3,getActivity().getResources().getColor(R.color.color3));
            }
        });
    }

    private void setColor(int selection,int color) {
        ccp.setContentColor(color);

        //textView
        textViewTitle.setTextColor(color);

        //editText
        editTextPhone.setTextColor(color);
        editTextPhone.setHintTextColor(color);
        editTextPhone.getBackground().setColorFilter(color, PorterDuff.Mode.SRC_ATOP);

        //remove selected bg
        resetBG();

        //set selected bg
        int selectedBGColor=getActivity().getResources().getColor(R.color.selectedTile);
        switch (selection){
            case 1:
                relativeColor1.setBackgroundColor(selectedBGColor);
                break;
            case 2:
                relativeColor2.setBackgroundColor(selectedBGColor);
                break;
            case 3:
                relativeColor3.setBackgroundColor(selectedBGColor);
                break;
        }
    }

    private void resetBG() {
        relativeColor1.setBackgroundColor(getActivity().getResources().getColor(R.color.dullBG));
        relativeColor2.setBackgroundColor(getActivity().getResources().getColor(R.color.dullBG));
        relativeColor3.setBackgroundColor(getActivity().getResources().getColor(R.color.dullBG));
    }

    private void assignViews() {
        textViewTitle =(TextView)getView().findViewById(R.id.textView_title);
        editTextPhone =(EditText)getView().findViewById(R.id.editText_phone);
        ccp=(CountryCodePicker)getView().findViewById(R.id.ccp);
        relativeColor1=(RelativeLayout)getView().findViewById(R.id.relative_color1);
        relativeColor2=(RelativeLayout)getView().findViewById(R.id.relative_color2);
        relativeColor3=(RelativeLayout)getView().findViewById(R.id.relative_color3);
    }
}

File: app/src/main/java/in/hbb20/countrycodepickerproject/CustomColorFragment.kt
package `in`.hbb20.countrycodepickerproject

import android.graphics.PorterDuff
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.core.content.ContextCompat.getColor
import androidx.fragment.app.Fragment
import com.hbb20.CountryCodePicker

/**
 * A simple [Fragment] subclass.
 */
class CustomColorFragment: Fragment() {

	private lateinit var buttonNext: Button
	private lateinit var textViewTitle: TextView
	private lateinit var editTextPhone: EditText
	private lateinit var ccp: CountryCodePicker
	private lateinit var relativeColor1: RelativeLayout
	private lateinit var relativeColor2: RelativeLayout
	private lateinit var relativeColor3: RelativeLayout
	private var selectedColor = -1

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		// Inflate the layout for this fragment
		return inflater.inflate(R.layout.fragment_custom_color, container, false)
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)

		assignViews()
		setClickListener()
	}

	private fun setClickListener() {
		relativeColor1.setOnClickListener { setColor(1, getColor(context!!, R.color.color1)) }

		relativeColor2.setOnClickListener { setColor(2, getColor(context!!, R.color.color2)) }

		relativeColor3.setOnClickListener { setColor(3, getColor(context!!, R.color.color3)) }

		buttonNext.setOnClickListener { (activity as ExampleActivity).viewPager.currentItem = (activity as ExampleActivity).viewPager.currentItem + 1 }

		ccp.setOnClickListener {
			ccp.launchCountrySelectionDialog()
			if (selectedColor != -1) {
				ccp.contentColor = selectedColor
			}
		}
	}

	private fun setColor(selection: Int, color: Int) {
		ccp.contentColor = color
		//textView
		textViewTitle.setTextColor(color)

		//editText
		editTextPhone.setTextColor(color)
		editTextPhone.setHintTextColor(color)
		editTextPhone.background.setColorFilter(color, PorterDuff.Mode.SRC_ATOP)

		//remove selected bg
		resetBG()

		//set selected bg
		val selectedBGColor = getColor(context!!, R.color.selectedTile)
		when (selection) {
			1 -> relativeColor1.setBackgroundColor(selectedBGColor)
			2 -> relativeColor2.setBackgroundColor(selectedBGColor)
			3 -> relativeColor3.setBackgroundColor(selectedBGColor)
		}

	}

	private fun resetBG() {
		relativeColor1.setBackgroundColor(getColor(context!!, R.color.dullBG))
		relativeColor2.setBackgroundColor(getColor(context!!, R.color.dullBG))
		relativeColor3.setBackgroundColor(getColor(context!!, R.color.dullBG))
	}

	private fun assignViews() {
		textViewTitle = view!!.findViewById(R.id.textView_title)
		editTextPhone = view!!.findViewById(R.id.editText_phone)
		ccp = view!!.findViewById(R.id.ccp)
		relativeColor1 = view!!.findViewById(R.id.relative_color1)
		relativeColor2 = view!!.findViewById(R.id.relative_color2)
		relativeColor3 = view!!.findViewById(R.id.relative_color3)
		buttonNext = view!!.findViewById(R.id.button_next)
	}
}
